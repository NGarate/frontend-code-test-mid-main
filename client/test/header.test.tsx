import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/header';
import CartContext from '../lib/cartContext';

const FakeProvider = ({
  children,
  initialCartItems,
}: {
  children: React.ReactNode;
  initialCartItems: CartItemsMap;
}) => {
  const [cartItems, setCartItems] = useState<CartItemsMap>(initialCartItems || new Map());
  const [lastUpdated, setLastUpdated] = useState<number>(Date.now());

  return (
    <CartContext.Provider
      value={{
        cartItems,
        setCartItems,
        lastUpdated,
        setLastUpdated,
      }}
    >
      {children}
    </CartContext.Provider>
  );
};

test('renders the Header correctly with no cart items', async () => {
  const tree = renderer
    .create(
      <FakeProvider>
        <Header />
      </FakeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('renders the Header correctly with multiple cart items', async () => {
  const customCartItems = new Map();
  customCartItems.set(1, 3);
  customCartItems.set(2, 1);
  customCartItems.set(3, 4);

  const tree = renderer
    .create(
      <FakeProvider initialCartItems={customCartItems}>
        <Header />
      </FakeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
