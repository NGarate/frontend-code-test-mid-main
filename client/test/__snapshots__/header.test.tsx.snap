// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header correctly with multiple cart items 1`] = `
<header
  className="sc-967ce249-0 gpbTCD"
>
  <div
    className="sc-967ce249-1 iihHil"
  >
    <img
      alt="Octopus Energy logo"
      className="sc-967ce249-2 kyYZFM"
      src="/octopus-logo.svg"
    />
    <button
      className="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-IconButton sc-a100302e-0 bFlczo"
      data-accent-color=""
      onClick={[Function]}
    >
      <img
        alt="Cart icon"
        src="/basket.svg"
      />
      <div
        className="sc-a100302e-1 kEMDru"
      >
        8
      </div>
    </button>
  </div>
</header>
`;

exports[`renders the Header correctly with no cart items 1`] = `
<header
  className="sc-967ce249-0 gpbTCD"
>
  <div
    className="sc-967ce249-1 iihHil"
  >
    <img
      alt="Octopus Energy logo"
      className="sc-967ce249-2 kyYZFM"
      src="/octopus-logo.svg"
    />
    <button
      className="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-IconButton sc-a100302e-0 bFlczo"
      data-accent-color=""
      onClick={[Function]}
    >
      <img
        alt="Cart icon"
        src="/basket.svg"
      />
      <div
        className="sc-a100302e-1 hbArcs"
      >
        0
      </div>
    </button>
  </div>
</header>
`;
